### ISSUE TITLE
[documentation] [keybindings] [C-c C-d] f , d

### ISSUE THREAD

**Issue by rprimus** [joaotavora/sly#172]
Fri Jul  6 20:15:22 BST 2018

[Documentation Commands](https://joaotavora.github.io/sly/#Documentation)

`sly-describe-symbol` and `sly-describe-function` are **only** available with the final key (d, f respectively) used with control.

This (so far), does not match the statements made in [Keybindings](https://joaotavora.github.io/sly/#Keybindings)
> We also have the following somewhat unusual convention of our own: when entering a three-key sequence, the final key can be pressed either with control or unmodified.

All of the other documentation commands are as listed in the manual, however,  they do not have a single letter equivalent (after using prefix `C-c C-d`.

#### Background

I am familiarising myself with `sly` and am going through the manual, so am raising issues as I come across them.

**Comment by joaotavora** [comment]
> I am familiarising myself with sly and am going through the manual, so am raising issues as I come across them.

That's great, most welcome!  I never really proof read the manual properly.

**Comment by rprimus** [comment]
Sat Jul  7 08:44:10 BST 2018

Will use here as a placeholder for all documentation related issues.

Merging [173]( https://github.com/joaotavora/sly/issues/172) here.

**Comment by rprimus** [comment]
Fri Jul  6 21:03:39 BST 2018

##  "C-c ~" and "C-c M-p"
In [5.8 Abort/Recovery Commands](https://joaotavora.github.io/sly/#Recovery):

- `C-c ~` is stated as being bound to `sly-sync-package-and-default-directory`.
This should state it is bound to `sly-mrepl-sync`.
- `C-c M-p` is unbound

**Comment by rprimus** [comment]
Sat Jul  7 08:52:06 BST 2018

### `C-c C-z` `M-x sly-mrepl`

[6.1 SLY REPL: the “top level”](https://joaotavora.github.io/sly/#REPL)

There are two entries in the manual (separated by the entry for `M-x sly-mrepl-new`) with descriptions:

- Start or select an existing main REPL buffer.
- Select the output buffer, preferably in a different window.

**Comment by rprimus** [comment]
Sat Jul  7 09:30:34 BST 2018

### SLYNK:*GLOBALLY-REDIRECT-IO*

[6.1.2 REPL output](https://joaotavora.github.io/sly/#REPL-output)
```
Symbol "*GLOBALLY-REDIRECT-IO*" not found in the SLYNK package.
```

**Comment by rprimus** [comment]
Sat Jul  7 09:44:20 BST 2018

[6.1.3 REPL backreferences](https://joaotavora.github.io/sly/#REPL-backreferences)

Last sentence:

> And that, in turn, is very is often as easy 

Extra `is`?

**Comment by joaotavora** [comment]
Yes, @rprimus, there are very many such inconsistencies.  I suggest that you make a PR and fix all of them as best as you can.  We can go over the result in a GitHub review before merging.  Or might be easier than opening a separate issue for each problem you find.  Thanks again for working on this.

**Comment by joaotavora** [comment]
> Will use here as a placeholder for all documentation related issues.

Good idea.

**Comment by rprimus** [comment]
Sat Jul  7 10:15:54 BST 2018
### [6.2 The Inspector](https://joaotavora.github.io/sly/#Inspector)

### `d`, `M-x sly-inspector-describe`


This should be replaced with:
`D`, `M-x sly-inspector-describe-inspectee`
-=-

### `p`, `M-x sly-inspector-pprint`
###  `.`,  `M-x sly-inspector-show-source`
These should be removed
-=-

### `h`, `M-x sly-inspector-history`
Needs to be added
-=-

### `M-RET`, `M-x sly-inspector-copy-down`
Replace function with
`sly-mrepl-copy-part-to-repl`
-=-

### `TAB`, ` M-x sly-inspector-next-inspectable-object`
### `S-TAB`,  `M-x sly-inspector-previous-inspectable-object`
Replace functions with `forward-button`, `backward-button` respectively
-=-

**Comment by rprimus** [comment]
Sun Jul  8 12:30:15 BST 2018

- [ ] 6.3.1 Examining frames
Mention that `D` `sly-db-disassamble` is not implemented for `sbcl`
- [ ] 6.3.2 Invoking restarts
- [ ] 6.3.3 Navigating between frames 
- [ ] 6.3.4 Stepping 
- [ ] 6.3.5 Miscellaneous Commands
 
- [ ] 7.1.1 Keybindings
- [ ] 7.1.2 Keymaps
- [ ] 7.1.3 Defcustom variables
- [ ] 7.1.4 Hooks

- [ ] 7.2.1 Communication style
- [ ] 7.2.2 Other configurables

- [ ] 8.1.1 Setting up the Lisp image
- [ ] 8.1.2 Setting up Emacs
- [ ] 8.1.3 Setting up pathname translations

- [ ] 8.2 Loading Slynk faster
- [ ] 8.3 Connecting to SLY automatically

- [ ] 9.1 Loading and unloading “on the fly”

- [ ] 9.2.1 TRAMP
- [ ] 9.2.2 Scratch Buffer

**Comment by joaotavora** [comment]
In general, feel free to make substantial changes if you think that it improves readability

> worth having a use-package example config somewhere in the manual?

I'm not a fan, but it's popular, so add it to the installation section, if it's not too long.

> Inconsistent WRT display of keybindings: @kbdanchorc used instead of @kdb

Actually, it should display the same way, but I don't mind if you get rid of the @kbdanchorc macro by expanding it inline.

>    generates error: Wrong type argument: integer-or-marker-p, nil
 >   if backtrace lists more, remaining frames are not displayed

Are these bugs? If so, make issues, please.

**Comment by joaotavora** [comment]
Also, I've merged what we've done in `update-docs` so far, but you can continue adding to that branch, and I'll merge progressively.

**Comment by joaotavora** [comment]
A tangent:

> http://www.catb.org/esr/faqs/smart-questions.html

"...also, you're worthless"  

Some useful advice in there despite the bullish urge to belittle others.

> http://www.chiark.greenend.org.uk/~sgtatham/bugs.html

Much better.

I also once wrote this on the matter (https://github.com/joaotavora/yasnippet#important-note-regarding-bug-reporting)

**Comment by rprimus** [comment]
Thu May  2 10:53:41 BST 2019

Hi @joaotavora - /me back to learning lisp (again).
Am going through `Marco Baringer's SLIME Tutorial Video` and noticed SLIME uses `C-c C-d h` for `Hyperspec Lookup`.

Using `C-h b`,  there are only three bindings listed for `hyperspec`.  In `SLY`,
`C-c C-d C-h` is used for `sly-documentation-lookup`.

It may be helpful (for newbies) to alias (or rename) this to `sly-hyperspec-lookup` - so it's more easily found:
    - For people coming from `SLIME`
    - grouped with other `hyperspec` commands (listed in `describe-bindings`)

Thoughts?

P.S.   Will be completing this PR  RSN!

**Comment by guicho271828** [comment]
> "C-c C-d f" and "C-c C-d d"
I am migrating from slime to sly recently, it is annoying every time I hit C-c C-d h (hyperspec lookup) to get nothing and retry with C-c C-d C-h. I wish this gets fixed soon.
